xapian-core/api/valuerangeproc.cc

```diff
@@ -471,13 +471,12 @@ UnitRangeProcessor::operator()(const string& b, const string& e)
 	    // Not a valid byte unit
 	    goto not_our_range;
 	}
-	string b_ = string(b, 0, b.size() - 1);
 
 	errno = 0;
-	const char * startptr = b_.c_str();
+	const char * startptr = b.c_str();
 	char * endptr;
 	num_b = strtod(startptr, &endptr);
-	if (endptr != startptr + b_.size() || errno) {
+	if (endptr != startptr + b.size() - 1 || errno) {
```

Comment:
For clarity, the '-1' is because endptr points to the last character of b, which could be among the byte units "B", "K", etc. I forgot the comment here but soon after, this line of code is modified in another commit.